### PR TITLE
[USMON-776] Fix dynamic table map cleaner interval 

### PR DESCRIPTION
**Pull request by amitslavin**
### What does this PR do?

Changed the http2 map cleaner interval configuration argument.

### Motivation

Changed the HTTP/2 map cleaner interval configuration value from `HTTPMapCleanerInterval` to `HTTP2DynamicTableMapCleanerInterval` in order to control the HTTP/2 map cleaner interval with a different configuration value dedicated for HTTP/2.

### CODE REVIEW

**Review by richl-dd — APPROVED** [review]

**Review by hmahmood — APPROVED** [review]

File: pkg/network/protocols/http2/protocol.go [hunk 1]
```diff
@@ -385,13 +385,17 @@ func (p *Protocol) setupHTTP2InFlightMapCleaner(mgr *manager.Manager) {
...
 			return (now - updated) > ttl
 		}
 
-		started := int64(val.Stream.Request_started)
-		return started > 0 && (now-started) > ttl
+		if started := int64(val.Stream.Request_started); started > 0 {
+			return (now - started) > ttl
+		}
+
+		// If we've seen only responses, then remove from the map the half-stream.
```
Comment (guyarb) [inline pkg/network/protocols/http2/protocol.go:397]: we should work on the documentation here

File: pkg/network/protocols/http2/protocol.go [hunk 2]
```diff
@@ -394,7 +394,8 @@ func (p *Protocol) setupHTTP2InFlightMapCleaner(mgr *manager.Manager) {
 			return (now - started) > ttl
 		}
 
-		// If we've seen only responses, then remove from the map the half-stream.
+		// In a case where the status code is finalized, that means that we managed to capture the response.
```
[the remark this answers is not in the document]
Comment (guyarb) [reply pkg/network/protocols/http2/protocol.go:397]: As we talked by heart, we can omit this specific change thanks to PR #22995

**Review by guyarb — APPROVED** [review]

**Comment by amitslavin** [comment]
I have changed the PR to use only the HTTP2DynamicTableMapCleanerInterval, which is safer as it is configurable.

**Comment by brycekahle** [comment]
@amitslavin can you update the PR description to match your changes?

**Comment by amitslavin** [comment]
> @amitslavin can you update the PR description to match your changes?

Changed, thanks for that!

**Pull request merged**